core/state_processor.go

```diff
@@ -1146,27 +1187,26 @@ func ProcessQiTx(tx *types.Transaction, chain ChainContext, checkSig, isFirstQiT
 		str := fmt.Sprintf("total value of all transaction inputs for "+
 			"transaction %v is %v which is less than the amount "+
 			"spent of %v", tx.Hash(), totalQitIn, totalQitOut)
-		return nil, nil, errors.New(str), nil
+		return nil, nil, nil, errors.New(str), nil
 	}
 
 	// the fee to pay the basefee/miner is the difference between inputs and outputs
 	txFeeInQit := new(big.Int).Sub(totalQitIn, totalQitOut)
+	qiTxFee = new(big.Int).Set(txFeeInQit)
```

Comment:
Why is there qiTxFee and txFeeInQit? They appear to be the same

Reply:
Its not the same, its used as the fee for conversion and gets subtracted, by i want the fee that is defined in the trasaction